Capture required args and make it match cli

tevoinea opened the pull request:
## Summary of the Pull Request

_What is this about?_

Closes #3427 

The `libfuzzer-fuzz` local task has the following required arguments:

```shell
execute a local-only libfuzzer fuzzing task

Usage: onefuzz-task local libfuzzer-fuzz [OPTIONS] --target_exe <target_exe> --inputs_dir <inputs_dir> --crashes_dir <crashes_dir> --crashdumps_dir <crashdumps_dir>
```

This PR updates our libfuzzer-fuzz basic yaml template to those required arguments. It also adds one more, `coverage` argument.

chkeita commented:
My initial comment was about matching the onefuzz cli template command. So, we also need the regression and analysis tasks in the template.

tevoinea commented:
> My initial comment was about matching the onefuzz cli template command. So, we also need the regression and analysis tasks in the template.

Ah I misunderstood, updated to match the libfuzzer_basic cli template. I added the regression task and since the analysis task is optional, I included it but commented out. I also found a bug in a mismatch of the required args (`tools` was supposed to be optional so I fixed that).

Porges commented on src/agent/onefuzz-task/src/local/template.rs:
```diff
@@ -241,12 +241,10 @@ mod test {
+        if schema_str.replace('\n', "") != checked_in_schema.replace('\n', "") {
+            std::fs::write("src/local/new.schema.json", schema_str)
+                .expect("The schemas did not match but failed to write new schema to file.");
+            panic!("The checked-in local fuzzing schema did not match the generated schema. The generated schema can be found at src/local/new.schema.json");
```
We could also use `insta` for this test, which can do JSON diffs and other nice things
